kafka/pykafka/balancedconsumer.py

```diff
@@ -0,0 +1,371 @@
+import itertools
+import logging as log
+import signal
+import socket
+import sys
+import time
+from uuid import uuid4
+import weakref
+
+from kazoo.client import KazooClient
+from kazoo.exceptions import NoNodeException, NodeExistsError
+from kazoo.recipe.watchers import ChildrenWatch
+
+from kafka.common import OffsetType
+from kafka.pykafka.simpleconsumer import SimpleConsumer
+
+
+class BalancedConsumer():
+    def __init__(self,
+                 topic,
+                 cluster,
+                 consumer_group,
+                 fetch_message_max_bytes=1024 * 1024,
+                 num_consumer_fetchers=1,
+                 auto_commit_enable=False,
+                 auto_commit_interval_ms=60 * 1000,
+                 queued_max_messages=2000,
+                 fetch_min_bytes=1,
+                 fetch_wait_max_ms=100,
+                 refresh_leader_backoff_ms=200,
+                 offsets_channel_backoff_ms=1000,
+                 offsets_commit_max_retries=5,
+                 auto_offset_reset=OffsetType.LATEST,
+                 consumer_timeout_ms=-1,
+                 rebalance_max_retries=5,
+                 rebalance_backoff_ms=2 * 1000,
+                 zookeeper_connection_timeout_ms=6 * 1000,
+                 zookeeper_connect='127.0.0.1:2181'):
```

Comment:
Instead of zk hosts, would it make more sense to take a kazoo instance as an argument?  It'd let the user change any settings in zk that they wanted and make our use easier. It's how samsa currently works, fwiw.